networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py

```diff
@@ -118,17 +126,23 @@ def _allocate_vxlan_segment(self, network, host_id, level, host_config):
             alloc = session.query(AllocationsModel).filter_by(host=host_id, level=level, segment_type=segment_type,
                                                               network_id=network_id).first()
             if alloc:
-                # check if segment is existing, if not the allocation should be deleted and a new one allocated
-                segment = (session.query(ml2_models.NetworkSegment).filter_by(id=alloc.segment_id).first())
+                # check if segment exists, if not the allocation should be deleted and a new one allocated
+                if alloc.segment_id is not None:
+                    segment = (session.query(ml2_models.NetworkSegment).filter_by(id=alloc.segment_id).first())
+                else:
+                    # segment was null in allocations table
+                    segment = None
+
                 if segment:
                     return alloc
                 else:
                     # TODO : handle case when allocation is on a segment no longer configured for use
                     # Or we can leave until 'natural' cleanup happens
                     alloc.update({"network_id": None, "segment_id": None})
```

Comment:
Not needed anymore I assume

Reply:
We force the segment_id to NULL with the ForeignKey `ondelete=SET NULL`. We also change the queries to use a NULLed `segment_id` as indicator for an unused row. So I think you're right, the removal of the network_id is purely cosmetic by now.

Reply:
Right, with the referential integrity we can just check if there is a segment_id set. If it is, then the alloc is valid.